### PR TITLE
Add feature that offers to convert from a conditional expression to ?.  

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
tagging @dotnet/roslyn-ide    Still need to do:  
1. UI
2. VB side
3. tests

**Comment by CyrusNajmabadi** [comment]
This branches off of https://github.com/dotnet/roslyn/pull/14816   Once that goes in, this will be a lot simpler.

File: src/Features/Core/Portable/UseNullPropagation/AbstractUseNullPropagationDiagnosticAnalyzer.cs [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+        protected abstract bool IsNotEquals(TBinaryExpressionSyntax condition);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.RegisterSyntaxNodeAction(AnalyzeSyntax, GetSyntaxKindToAnalyze());
+        }
+
+        private void AnalyzeSyntax(SyntaxNodeAnalysisContext context)
+        {
+            var conditionalExpression = (TConditionalExpressionSyntax)context.Node;
```
Comment (jbevain) [inline src/Features/Core/Portable/UseNullPropagation/AbstractUseNullPropagationDiagnosticAnalyzer.cs]: The diagnostic needs to be restricted to C# 6 and above and VB 14.

**Comment by CyrusNajmabadi** [comment]
Option checking added.

File: src/Workspaces/Core/Portable/Editing/SyntaxGenerator.cs [hunk 2]
```diff
@@ -1728,6 +1728,10 @@ public SyntaxNode DottedName(string dottedName)
         /// </summary>
         public abstract SyntaxNode ConditionalExpression(SyntaxNode condition, SyntaxNode whenTrue, SyntaxNode whenFalse);
 
+        internal abstract SyntaxNode ConditionalAccessExpression(SyntaxNode expression, SyntaxNode whenNotNull);
```
Comment (jmarolf) [inline src/Workspaces/Core/Portable/Editing/SyntaxGenerator.cs]: >internal [](start = 8, length = 8)

Why can't these be public?

**Comment by jmarolf** [comment]
:shipit:

File: src/Workspaces/Core/Portable/Editing/SyntaxGenerator.cs (hunk 2 above)
Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/Editing/SyntaxGenerator.cs]: I don't want to have to go through the public API process :)

**Pull request merged**